### PR TITLE
Orion Express Courier System

### PR DESCRIPTION
**Pull request by Geevies**
* Added the Orion Express Courier System, a new gameplay loop for Hangar Techs. Orion Express packages will now spawn in the hangar that are linked to delivery points on exoplanets, delivering these packages will provide the cargo department with some funds, depending on the package.

This PR has some goodies not mentioned in the changelog that I would like players to discover organically via gameplay, the extended descriptions on the items and objects should be enough to get them well on their way. Feel free to look if you're code literate, but I ask that you don't spoil some of the bonus functionality not listed anywhere here.

With that out of the way...

### Welcome to the Orion Express Courier System!

Orion Express packages, painted white and black, will now spawn in the Horizon's warehouse should a valid exoplanet and delivery point combo be found. I've coded it in such a way that all exoplanets located within corporate sectors will have a delivery point, but that may change in the future.

To get started, a hangar technician will travel down into the warehouse, to the back, and open the new courier's locker, where they will find all* the supplies they'll need to complete a delivery. This includes the newly added cargo pack, which can hold up to two cargo packages on the courier's back.

After the courier has suited up and collected their packages (bringing a friend is advised, those packages are heavy!), they can make their way to whoever is willing to give them a ride, be it a Miner on the Spark, a Xenoarchaeologist on the Intrepid, or a bored Bridge Crew member on the Canary, who will transport them to the exoplanet marked on the package.

After landing, the courier will make their way to** the delivery point, load the packages into the chute, and be graciously rewarded*** for their efforts.

And that's it! Easy peasy.

\* The bare minimum required to keep a courier alive on the surface of an exoplanet.
** Fight through hostile xenofauna and space radiation.
*** Couriers receive a 2% tip on top of each delivery, while the cargo department get all the funds listed on the package. If a courier doesn't have an ID on them, the tip will be printed as cash. Be aware that unauthorized personnel may be able to intercept a package and claim the tip for themselves, always bring a friend.

### Screenshots
31wM8b1ARb
image
image

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: code/modules/cargo/delivery/backpack.dm [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	contained_sprite = TRUE
+	w_class = ITEMSIZE_HUGE
+	slot_flags = SLOT_BACK
+	drop_sound = 'sound/items/drop/backpack.ogg'
+	pickup_sound = 'sound/items/pickup/backpack.ogg'
+
+	var/list/contained_packages
+
+/obj/item/cargo_backpack/Destroy()
+	QDEL_NULL(contained_packages)
```
Comment (NonQueueingMatt) [inline code/modules/cargo/delivery/backpack.dm:17]: you are QDEL_NULLing a list here, need to use QDEL_NULL_LIST

File: code/modules/cargo/delivery/backpack.dm [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	return ..()
+
+/obj/item/cargo_backpack/examine(mob/user, distance)
+	. = ..()
+	if(length(contained_packages))
+		for(var/obj/item/cargo_package/package in contained_packages)
+			to_chat(user, SPAN_NOTICE(" - SITE: <b>[package.delivery_point_sector]</b> | COORD: <b>[package.delivery_point_coordinates]</b> | ID: <b>[package.delivery_point_id]</b>"))
```
Comment (NonQueueingMatt) [inline code/modules/cargo/delivery/backpack.dm:24]: may be worth making this a small tgui opened on examine

File: code/modules/cargo/delivery/package.dm [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+		delivery_point_coordinates = "[delivery_point.x]-[delivery_point.y]"
+
+/obj/item/cargo_package/examine(mob/user, distance)
+	. = ..()
+	to_chat(user, SPAN_NOTICE("The label on the package reads: SITE: <b>[delivery_point_sector]</b> | COORD: <b>[delivery_point_coordinates]</b> | ID: <b>[delivery_point_id]</b>"))
+	to_chat(user, SPAN_NOTICE("The price tag on the package reads: <b>[pay_amount]电</b>"))
+
+/obj/item/cargo_package/do_additional_pickup_checks(var/mob/living/carbon/human/user)
+	if(!ishuman(user))
+		to_chat(user, SPAN_WARNING("\The [src] is way too heavy for you to pick up!"))
```
Comment (NonQueueingMatt) [inline code/modules/cargo/delivery/package.dm:39]: this would be a bit of a weird message for a borg to get, I'd probably just remove it

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

File: code/modules/cargo/delivery/backpack.dm (hunk 2 above)
Comment (Geevies) [reply code/modules/cargo/delivery/backpack.dm]: added, looks like this:
image

**Comment by Geevies** [comment]
!review

**Review by DreamySkrell — APPROVED** [review]
our first strand-type PR

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**